### ISSUE TITLE
Use cache.compute() method to improve the replace retry loop

### ISSUE THREAD

**Issue by pruivo** [keycloak/keycloak#29073]
### Description

The `Map.replace()` returns a boolean to signal if the replaced succeed or not. This is inefficient because KC needs to perform a read operation to fetch the value again.

Using the `compute()` method, we can retrieve the most recent version and we can do the replacement based on version instead of equality.

**Comment by ahus1** [comment]
I like to backport this change to KC24 as it IMHO fixed #28466. As this is a breaking change and it would 

@pruivo - please create a first backport which would allow us in a two-step approach to get all necessary classes in a first patch release. Once this is merged and released, there would be a second PR to actually use these classes, IMHO that would be the changes to InfinispanChangelogBasedTransaction.

**Comment by pruivo** [comment]
Pull Request to KC24 available here: https://github.com/keycloak/keycloak/pull/29229

**Comment by ahus1** [comment]
On hold for 24.0.4 to be released. Then the second half will be backported.

**Comment by ahus1** [comment]
@pruivo - KC 24.0.4 has been released today. Please provide the second part of the backport. Thanks!

**Comment by pruivo** [comment]
second part here: https://github.com/keycloak/keycloak/pull/29378

**Comment by xgp** [comment]
This seems to have _caused_ the "Failed to replace entity" problem when upgrading to `24.0.5`. See https://github.com/keycloak/keycloak/issues/30451
